crates/goose-mcp/src/tutorial/tutorials/build-mcp-extension.md

```diff
@@ -0,0 +1,333 @@
+# Building an Extension with MCP
+
+For this tutorial you will guide the user through building an MCP extension.
+This will require you to get familiar with one of the three available SDKs: 
+Python, TypeScript, or Kotlin.
+
+MCP extensions allow AI agents to use tools, access resources, and other more advanced
+features via a protocol. The extension does not need to include all of these features. 
+
+## Your Role
+- You will help users implement MCP extensions using their chosen SDK
+- You should adapt your explanations based on the user's experience level and questions
+- Always reference the SDK implementations for accurate, up-to-date details
+
+## Initial Setup
+
+You (the agent) should run the following so that you can get an up to date
+reference of the SDK to refer to.
+
+Clone the SDK repo into a temp dir:
+
+```bash
+mkdir -p /tmp/mcp-reference && cd /tmp/mcp-reference
+git clone https://github.com/modelcontextprotocol/[python|typescript|kotlin]-sdk.git
+```
```

Comment:
i noticed that it didn't clone the reference sdk unless i specifically asked it to, it didn't matter as much for python but telling it to check the reference helped it with kotlin